Source/Scene/HereMapsImageryProvider.js

```diff
@@ -0,0 +1,521 @@
+     *.       are available.
+     * @param {String} [options.tileType="maptile"] Determines which type of image will be delivered. E.g. Map, Labels, Streets, etc.
+     * @param {String} [options.scheme="satellite.day"] Specifies the view scheme, e.g normal.day, normal.night, terrain.day, etc.
+              Note, some schemes will be rejected by certain baseUrls.
+     * @param {String} [options.tileFormat="jpg"] Valid values are jpg, png, and png8.
+     * @param {Number} [options.tileSize=256] The width and height of each tile in pixels. Supported values are 256 and 512.
+     * @param {Boolean} [options.forceUseNewest=false] By default, the current map version will be queried only once at start.
+     *        Subsequent tile requests will always use a hash of this version so that tiles are consistent in case the map is updated
+     *        while the app is running. Enabling this flag will cause all tiles to always reqeuest the newest version.
+     *        See https://developer.here.com/documentation/map-tile/topics/tile-version.html#tile-version for details.
+     *
+     *
+     * @see ArcGisMapServerImageryProvider
+     * @see GoogleEarthImageryProvider
+     * @see OpenStreetMapImageryProvider
+     * @see SingleTileImageryProvider
+     * @see TileMapServiceImageryProvider
+     * @see WebMapServiceImageryProvider
+     *
+     * @see {@link https://developer.here.com/documentation/map-tile/topics/overview.html}
+     * @see {@link http://www.w3.org/TR/cors/|Cross-Origin Resource Sharing}
+     *
+     * @example
+     * var here = new Cesium.HereMapsImageryProvider({
+     *     url : 'aerial.maps.cit.api.here.com',
+     *     appId : 'get-yours-at-https://www.developer.here.com/',
+     *     appCode : 'get-yours-at-https://www.developer.here.com/',
+     * });
+     */
+    var HereMapsImageryProvider = function HereMapsImageryProvider(options) {
+        options = defaultValue(options, {});
+
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(options.appId)) {
+            throw new DeveloperError('options.appId is required.');
+        }
+        if (!defined(options.appCode)) {
+            throw new DeveloperError('options.appCode  is required.');
+        }
+        //>>includeEnd('debug');
+
+        var that = this;
+
+        this._proxy = options.proxy;
+        this._credit = new Credit({
+            text: '2014 DigitalGlobe, Inc.',
+            imageUrl: HereMapsImageryProvider._logoData,
+            link: 'http://www.here.com'
+        });
+
+        this._tilingScheme = new WebMercatorTilingScheme();
+        this._tileSize = defaultValue(options.tileSize, 256);
+
+        this._baseUrl = defaultValue(options.baseUrl, 'aerial.maps.api.here.com');
+        var mapId = 'newest';
+
+        var baseResource = new Resource({
+            url: '//{subdomain}.' + this._baseUrl,
```

Comment:
This is an odd setup. We can change the URL, but all are required to have 4 subdomains?

Reply:
That's right. HereMaps forces us to always use one of the four subdomains. Which is why I chose not to accept a Resource or full url in the options object. We basically have to always construct the URL.

Reply:
What are the possible URLs? Are they all like `aerial.maps.api.here.com`? Do they always end in `maps.api.here.com`? I'm not sure of the cleanest way to handle this.

Reply:
There's documentation in baseUrl description in the constructor docs. Basically the url is *.maps.api.here.com, where * is one of <aerial|base|pano|traffic>. They've got good API documentation here: https://developer.here.com/documentation/map-tile/topics/request-constructing.html. 

The url construction is more complicated than Bing, and the *.maps.api subdomain that you pick determines a lot of the other settings that are viable. Because of this I chose to put more impetus on the user to read the docs and determine what options are usable, rather than try to enumerate all the options and their dependencies like BingMapsStyle. The documentation has links to all the HereMaps API documentation that a user might need. I hope this is an acceptable level of complexity for a Cesium user who wants to use Here. The default settings should cover the common case.

Reply:
Ok. So essentially the subdomain and type are part of the REST API even though they are in the domain, which is a little odd. What do you think of this? Take a `options.url` that is a string or resource (`http://maps.api.here.com`) and a `options.type`. Then from there you build the URL.

I think this is the best option, since that is how the URL is formatted. Also, for parsing the url we have a `ThirdParty/Uri` that might help.

Reply:
That scheme sounds good to me. ThirdParty/Uri looks nice, thanks for pointing me to it.